restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/DocumentsResolver.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.web.docsapi.service.query.search.resolver;
+
+import io.reactivex.rxjava3.core.Flowable;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.dao.Paginator;
+import io.stargate.web.docsapi.service.DocsApiConfiguration;
+import io.stargate.web.docsapi.service.RawDocument;
+
+/** Base interface for all the document resolvers. */
+public interface DocumentsResolver {
+
+  /**
+   * Returns documents.
+   *
+   * @param db Document DB
+   * @param configuration {@link DocsApiConfiguration}
+   * @param keyspace keyspace to search in
+   * @param collection collection to search in
+   * @param paginator {@link Paginator}
+   * @return Flowable of documents.
+   */
+  Flowable<RawDocument> getDocuments(
+      DocumentDB db,
```

Comment:
Can we use `QueryExecutor` directly instead of `DocumentDB`? This should also allow us to unify authN and authZ... Some `DocumentDB` "query" methods run auth checks, but others do not :thinking:

Reply:
This could be possible, however, we are depending as well on the:

- query builder
- data store prepare query

and even if we would have all of those, we would end up in most likely duplicating the code in the `DocumentDB`.. Maybe we should solve the auth problem at the right place ->  `DocumentDB`?

Reply:
Actually, maybe we should really move to the `QueryExecutor`, this would help testing so much.. so how to solve the query builder and preparation, should we expose the data store from the `QueryExecutor.getDataStore()`

Reply:
Yes, that would be fine :+1: